README.md

```diff
@@ -9,25 +9,60 @@ This experiment comes in two phases:
 ## Generate Blackbox data
 
 You will need to ask the BlueJ developers for a Blackbox account. This comes
-with various conditions which you must adhere to.
+with various conditions which you must adhere to. The basic process is that you
+need to either generate a fresh, or use an existing, `blackbox/combos` file.
+From that, Java source files will be written to `blackbox/src_files`. **The
+latter must not be redistributed under any circumstances.** However, the
+`blackbox/combos` file is safe to redistribute.
 
-Once you have done that, the scripts in the `blackbox` directory can be used to
-generate Java source files from user input. Run `build_blackbox_data.sh` on
-the Blackbox server and, many hours later, you will have a `combos` file (which
-is the list of pairs needed to regenerate the source files) as well as the
-Java source in a `src_files` directory. **The latter must never be
-redistributed.** However, the `combos` file is safe to redistribute.
+
+### Using the same data we did in the paper
+
+If you are using a formal release of this project (i.e. you have not just
+cloned the repository), then you can recreate exactly the same Java source
+files we used via the following commands:
+
+```sh
+$ cd blackbox
+$ ./build_blackbox_data.sh
+```
+
+This will detect the existing `blackbox/combos` file and use that to recreate
+the same Java source files we used: those source files will be put into
+`blackbox/src_files`.
+
+
+### Generating fresh data
+
+If you want to generate a fresh batch of Blackbox data (or if you have just
+cloned this repository), 
```

Comment:
s/,/.

Reply:
I really messed that bit up! The new commit should fix this.